src/main/java/de/tum/in/www1/artemis/service/listeners/ResultListener.java

```diff
@@ -0,0 +1,62 @@
+package de.tum.in.www1.artemis.service.listeners;
+
+import javax.persistence.PostPersist;
+import javax.persistence.PostUpdate;
+import javax.persistence.PreRemove;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.stereotype.Component;
+
+import de.tum.in.www1.artemis.domain.Result;
+import de.tum.in.www1.artemis.service.ScoreService;
+
+@Component
+public class ResultListener {
+
+    @Value("#{new Boolean('${artemis.resultListener.isActive}')}")
+    private Boolean isActive;
+
+    private ScoreService scoreService;
+
+    /**
+     * While {@link javax.persistence.EntityManager} is being initialized it instantiates {@link javax.persistence.EntityListeners} including
+     * {@link ResultListener}. Now {@link ResultListener} requires the {@link ScoreService} which requires {@link de.tum.in.www1.artemis.repository.StudentScoreRepository}
+     * which requires {@link javax.persistence.EntityManager}. To break this circular dependency we use lazy injection of the service here.
+     *
+     * @param scoreService the student score service that will be lazily injected by Spring
+     */
+    public ResultListener(@Lazy ScoreService scoreService) {
+        this.scoreService = scoreService;
+    }
+
+    /**
+     * Remove or update associated participation scores before a result is removed
+     *
+     * @param resultToBeDeleted result about to be remove
+     */
+    @PreRemove
+    public void removeOrUpdateAssociatedParticipantScore(Result resultToBeDeleted) {
+        if (isActive == null || !isActive) {
```

Comment:
Why is this needed?

The problem was that during tests a lot of tests failed as after making a REST call the mock mvc removes the authentication object meaning no changing sql statements (create, update, delete) are possible any more. Thus the entity listener needs to be deactivated during tests.